### PR TITLE
fix: handle errors parsing files

### PR DESCRIPTION
**Pull request by kg6zvp**

### CODE REVIEW

File: src/reading/fromZip.js [hunk 1]
```diff
@@ -16,6 +16,11 @@ export default ({ fileFilter, parser=gdalParser }) => {
   unzip(inStream, { fileFilter })
     .then(({ files, done }) => {
       finished(out, done)
+      if (files.length < 1) {
```
Comment (stephenlacy) [inline src/reading/fromZip.js]: length === 0 is better as the files can't be -1

File: package.json [hunk 2]
```diff
@@ -203,6 +203,7 @@
     "mkdirp": "^0.5.1",
     "pumpify": "^2.0.0",
     "tempfile": "^3.0.0",
+    "through2": "^3.0.1",
```
Comment (stephenlacy) [inline package.json:206]: is this used?

File: src/reading/parseFile.js [hunk 3]
```diff
@@ -4,6 +4,8 @@ import mapValues from 'lodash.mapvalues'
 
 const wgs84 = gdal.SpatialReference.fromEPSG(4326)
 
+const maxLayers = 20
```
Comment (stephenlacy) [inline src/reading/parseFile.js:7]: is that a reasonable max, what happens if it is set to 100?

**Review by stephenlacy — CHANGES_REQUESTED** [review]

File: src/reading/fromZip.js (hunk 1 above)
Comment (kg6zvp) [reply src/reading/fromZip.js]: :+1:

File: package.json (hunk 2 above)
Comment (kg6zvp) [reply package.json:206]: Yes. I'm not sure how we were getting by without it except that vandelay already is using this.

File: src/reading/parseFile.js (hunk 3 above)
Comment (kg6zvp) [reply src/reading/parseFile.js:7]: The file that crashed only had like 30-something layers I believe, so I felt like 20 was a number that would work. If it's an issue, we can revisit it later.

File: test/inputs/gdb.js [hunk 4]
```diff
@@ -38,4 +38,14 @@ describe('from(gdb)', () => {
       }
     })
   })
+  it('should end stream early', async () => {
```
Comment (stephenlacy) [inline test/inputs/gdb.js]: This test does not actually test that the stream closes, it just increments a number, checks that number, and closes a stream.

**Review by stephenlacy — CHANGES_REQUESTED** [review]

File: src/reading/fromZip.js [hunk 5]
```diff
@@ -16,6 +16,11 @@ export default ({ fileFilter, parser=gdalParser }) => {
   unzip(inStream, { fileFilter })
     .then(({ files, done }) => {
       finished(out, done)
+      if (files.length === 0) {
```
Comment (yocontra) [inline src/reading/fromZip.js:19]: Shouldn't this just be `if (files.length === 0) return done()` one line up?

File: src/reading/parseFile.js (hunk 3 above)
Comment (yocontra) [inline src/reading/parseFile.js:7]: This seems like a hack - we should figure out why it crashed over 20 instead of limiting it to 20.

**Pull request merged**